src/librawspeed/metadata/CameraMetaData.h

```diff
@@ -23,18 +23,21 @@
 #include "common/Common.h" // for uint32
 #include <map>             // for map
 #include <string>          // for string
+#include <tuple>           // for tuple
 
 namespace RawSpeed {
 
 class Camera;
 
+using CameraId = std::tuple<std::string, std::string, std::string>;
+
```

Comment:
Ah, i forgot about `[std::tie](http://en.cppreference.com/w/cpp/utility/tuple/tie)`, then the `operator<` i was trying to avoid by using tuple is trivial, okay.

> Maybe you could merge the two efforts if you moved/renamed my TiffID struct and added a mode string there. It would unify the whole 'id' related information?

I did not look at that pr in *that detail*, so i have no clue what you are talking about. But since i'll need to pick this change to dt-2.2.x, this needs to be as non-intrusive as possible.